tests/integ/test_factorization_machines.py

```diff
@@ -53,3 +54,45 @@ def test_factorization_machines():
         assert len(result) == 10
         for record in result:
             assert record.label["score"] is not None
+
+
+def test_async_factorization_machines():
+
+    training_job_name = ""
+    endpoint_name = name_from_base('factorizationMachines')
+    sagemaker_session = sagemaker.Session(boto_session=boto3.Session(region_name=REGION))
+
+    with timeout(minutes=5):
+
+        data_path = os.path.join(DATA_DIR, 'one_p_mnist', 'mnist.pkl.gz')
+        pickle_args = {} if sys.version_info.major == 2 else {'encoding': 'latin1'}
+
+        # Load the data into memory as numpy arrays
+        with gzip.open(data_path, 'rb') as f:
+            train_set, _, _ = pickle.load(f, **pickle_args)
+
+        fm = FactorizationMachines(role='SageMakerRole', train_instance_count=1,
+                                   train_instance_type='ml.c4.xlarge',
+                                   num_factors=10, predictor_type='regressor',
+                                   epochs=2, clip_gradient=1e2, eps=0.001, rescale_grad=1.0 / 100,
+                                   sagemaker_session=sagemaker_session, base_job_name='test-fm')
+
+        # training labels must be 'float32'
+        fm.fit(fm.record_set(train_set[0][:200], train_set[1][:200].astype('float32')), wait=False)
+        training_job_name = fm.latest_training_job.name
+
+        print("Detached from training job. Will re-attach in 20 seconds")
+        time.sleep(20)
+        print("attaching now...")
+
+    with timeout_and_delete_endpoint_by_name(endpoint_name, sagemaker_session, minutes=35):
```

Comment:
35 minutes is too long. Can we make a test under 15 minutes?

Reply:
The test itself is taking the same time as the synchronous fit one, the difference is this timeout is accounting for fit() + deploy(). If you notice the timeout is consistent with the synchronous tests that we have.

The difference is the call to

with timeout_and_delete_endpoint_by_name() is spending a lot of time on attach()  this time is usually spent in fit() in the synchronous tests. In this case the fit() returns right away so it doesn't account for much of the runtime. So the test takes the same amount of time to complete.